Orleans/src/Microsoft.Orleans.Host.Abstractions/AzureStorageProvider.cs

```diff
@@ -0,0 +1,53 @@
+﻿using Microsoft.Extensions.DependencyInjection;
+using System;
+using System.Threading.Tasks;
+
+namespace Microsoft.Orleans.Host.Abstractions
+{
+    // this would go into the azure storage providers DLL
+    public static class AzureStorageProviderExtensions
+    {
+        public static void AddAzureStorage(this IStorageProvidersBuilder factory, string providerName, string connectionString)
+        {
+            var concreteFactory = ActivatorUtilities.CreateInstance<AzureStorageProviderFactory>(factory.Silo.ApplicationServices);
```

Comment:
feels like concreteFactory can be reused for creating other AzureStorageProvider. almost a singleton. maybe the method signature should be 
`public static void AddAzureStorage(this IStorageProvidersBuilder factoryBuilder, AzureStorageProviderFactory factory, string providerName, string connectionString) `